### PR TITLE
fix: Update the docs for governed documents

### PR DESCRIPTION
**Pull request by enikonovad**
Signed-off-by: Ekaterina Nikonova <enikonovad@gmail.com>

## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [ ] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [ ] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary
Updated the documentation for the authoring of the documents. 
Added collapsible menus. Updated the content to the current state of Trestle.

Link to the new page: https://github.com/IBM/compliance-trestle/blob/fix/update_govdocs/docs/trestle_author.md
## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: docs/trestle_author.md [hunk 1]
```diff
@@ -1,4 +1,13 @@
-# trestle CLI for governance of authored documents
+# Trestle CLI for Governance of Authored Documents
+
+This document is organized in a multiple collapsable sections for the ease of use.
+Start by reading the overview and setting up the Trestle workspace and then proceed to other sections,
```
Comment (fsuits) [inline docs/trestle_author.md:4]: for ease of use

File: docs/trestle_author.md [hunk 2]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
-In order for trestle to enforce structure an approach has been taken for how to template markdown documents. There are two mechanism that are enforced:
+To create a trestle workspace that will contain all your templates and governed documents:
 
-1. Enforcing a heading structure within the markdown document:
-1. Enforcing a structured header within the markdown document either by using yaml headers or a designated heading.
+```bash
+mkdir my_workspace
+cd my_workspace
+trestle init
+>>> Initialized trestle project successfully in [user_path]/my_workspace
```
Comment (fsuits) [inline docs/trestle_author.md:46]: update this with the new init modality

File: docs/trestle_author.md [hunk 3]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
+
+- <details>
+  <summary><b>Documents Structural Enforcement</b></summary>
+
+  ## Markdown structural enforcement
+
+  Trestle templating enforces the documents to follow a specific structure. There are two ways in which structure is enforced in a document against template:
+
+  1. Enforcing a header (metadata) structure within the markdown document.
+  1. Enforcing a heading structure within the markdown document.
```
Comment (fsuits) [inline docs/trestle_author.md:64]: header and heading is confusing.  Should you refer to "header" as "yaml header" for clarity?

File: docs/trestle_author.md [hunk 4]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
 
-```markdown
-# Template heading 1
+  The document YAML header(metadata) is said to be **_valid_** against the template, if and only if:
 
-# Template heading 2
-## Template sub heading
-```
+  1. It contains **all** keys from the template EXCEPT the keys that are either:
+     1. Start with `x-trestle`
```
Comment (fsuits) [inline docs/trestle_author.md:100]: keys that either

File: docs/trestle_author.md [hunk 5]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
-### Strict header / heading conformance mechanisms
+  The document body is said to be **_valid_** against the template, if and only if,
 
-Two mechanisms are provided to enforce metadata within markdown documents. The first is the yaml header, as used by technologies such as jekyll, the second is a markdown 'governed heading\` where templating of the content is enforced.
-Use of the yaml header is strongly encouraged as a first preference.
+  1. It contains **_all_** the headings provided in the template.
+  1. No new headings were added at the top level (i.e. `# New heading` is not allowed).
+  1. All original headings are in the same order as in the template.
+  1. All headings must be in the hierarchical order (i.e. `# Heading` then `### Heading` then `## Heading` is invalid).
+  1. If the `--governed-heading` option is provided, then
```
Comment (fsuits) [inline docs/trestle_author.md:137]: then... what?

File: docs/trestle_author.md [hunk 6]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
-- If the template heading text is wrapped in curly brackets `{}` then the name is not measured e.g. `# {Insert title here}`.
+  Markdown headings are considered to be nested based on the heading level (i.e. number of `#`). For example, `## 1.1 Lower level heading` is below `# 1. top level heading`.
 
-### Strict header / heading conformance mechanisms
+  The document body is said to be **_valid_** against the template, if and only if,
 
-Two mechanisms are provided to enforce metadata within markdown documents. The first is the yaml header, as used by technologies such as jekyll, the second is a markdown 'governed heading\` where templating of the content is enforced.
-Use of the yaml header is strongly encouraged as a first preference.
+  1. It contains **_all_** the headings provided in the template.
+  1. No new headings were added at the top level (i.e. `# New heading` is not allowed).
```
Comment (fsuits) [inline docs/trestle_author.md:133]: Can a heading be repeated in a template?  This isn't so much a question for the docs as an actual question about how it works.  If there are any issues with duplicate headings it should probably be mentioned.

File: docs/trestle_author.md [hunk 7]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
+  ```markdown
+  # Template heading 1
+  Some text
+  # Template heading 2
+  ## Template sub heading
+  ```
 
-For enforcing a governed heading the structural enforcement mechanism assumes that the `key:value` structure simply takes the form that following that for each line of content under the chosen heading the template content is a subset of the measured document, in the order provided in the template. This is performed after removing formatting (such as bolding), and any HTML comments.
+  Now we did added new sections and ended up with the following document.
+  This document will be valid against the template above
```
Comment (fsuits) [inline docs/trestle_author.md:148]: Then we added new sections...

File: docs/trestle_author.md [hunk 8]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
+  This sub-sub heading is okay
+  ```
 
-```markdown
-# heading for strict enforcement
-my_key:
-**my_key_2:**
-my_other key with strange stuff??
-```
+  However, violations such as adding or removing a heading at the top level is not acceptable:
```
Comment (fsuits) [inline docs/trestle_author.md:164]: are not acceptable

File: docs/trestle_author.md [hunk 9]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
-![Editing drawio data](assets/drawio_editing_data.png "Editing drawio data")
+  ![Editing drawio data](assets/drawio_editing_data.png "Editing drawio data")
 
-## Template versioning
+  </details>
 
-### Prerequisite:
+- <details>
+  <summary><b>Setting Up Templates and Documents Via CLI</b></summary>
+  Trestle allows to set up templates and governed documents in a several different ways based on the nature of the needed governance.
```
Comment (fsuits) [inline docs/trestle_author.md:197]: Trestle allows the setup of templates and ...

File: docs/trestle_author.md [hunk 10]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
+    ### Validating the documents against the template
 
-#### Header only validation (`-hov`, `--header-only-validate`)
+    To validate the documents against a template run:
 
-Turns off the validation of the structure of the document and only validates the yaml header structure.
+    > trestle author docs validate -tn my_task_name
 
-#### Ignore files or folders from validation (`-ig`, `--ignore`)
+    This will take this `TRESTLE_ROOT/.trestle/author/my_task_name/template.md` template and validate all markdown files here: `TRESTLE_ROOT/my_task_name/*.md`.
```
Comment (fsuits) [inline docs/trestle_author.md:265]: This will take the...

File: docs/trestle_author.md [hunk 11]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
 
-Provide regular expression after this flag to ignore folders and files which names match this regular expression during **validation only**.
+    This will validate that markdown body in the document is valid against the template. Please note that by default the header will **not** be validated.
```
Comment (fsuits) [inline docs/trestle_author.md:267]: Running the command will validate that...

File: docs/trestle_author.md [hunk 12]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
-trestle_root
- ┣ .trestle
- ┃ ┣ author
- ┃ ┃ ┗ my_task_2
- ┃ ┃ ┃ ┣ a_template.md
- ┃ ┃ ┃ ┣ another_template.md
- ┃ ┃ ┃ ┗ template.drawio
- ┃ ┗ config.ini
-```
+    `author folders` is designed to allow the assembly of groups of templates where the folder assembly is the unique instance. Trestle author folders supports validation of both markdown and drawio files. Note that headers / metadata must be specified in each applicable template.
```
Comment (fsuits) [inline docs/trestle_author.md:299]: Please expand on what you mean by "unique instance".  I think a few sentences are needed.

File: docs/trestle_author.md [hunk 13]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
+    - If `--template-version 1.0.0` (`-tv`) is passed the header field `x-trestle-template-version` will be ignored and document will be forcefully validated against template of version `1.0.0`.
+      Use this for testing purposes _only_ when you need to validate the document against a specific template. By default the template version will be determined based on `x-trestle-template-version` in the document.
+
+    </details>
+
+  - <details>
+      <summary><b>headers - Validate documents headers only against a global or a specific template </b></summary>
+
+    ## Validate documents headers only against a global or a specific template
+
```
Comment (fsuits) [inline docs/trestle_author.md:405]: This is hard to parse.  Do you mean: Validate only the document headers, and do it against a global or specific template.

I would avoid "documents headers" if you do mean "document headers" - and putting "only" first makes it clear only the headers are validated.

File: docs/trestle_author.md [hunk 14]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
+    In this section we describe the functionality of `trestle author headers` command.
+
+    Trestle author headers supports a slightly different usecase than that of `docs` and `folders` above: some content is governed, however, the content is non-standardized.
+
```
Comment (fsuits) [inline docs/trestle_author.md:409]: I think you mean "content is not standardized" - but I'm not sure what that means.  Can you add a sentence saying what you mean by not standardized.

File: docs/trestle_author.md [hunk 15]
```diff
@@ -18,347 +27,573 @@ While trestle provides editing support for OSCAL there is an unfortunate truth t
...
+    Please note that **only** headers will be validated. There is no option to validate the body.
 
-#### Template version tracking (`-tv`, `--template-version`)
+    ### Extra options
 
-This, along with the use of x-trestle-template-version in the governed header allows for repos using author templates
-to implement support for multiple versions of templates. The x-trestle-template-version header represents the version
-of the template used to create an instance document. With this change, the Version header specifically refers to the
-version _of that document_ (be it an instance or the template itself). This means:
+    - If `--global` (`-g`) is passed then `__global__` then it will create `trestle_root/.trestle/author/__global__` folder. Use this when you would like to use same template for multiple tasks.
```
Comment (fsuits) [inline docs/trestle_author.md:475]: I think there is a typo because there are two 'then's.  Not sure what `__global__` is doing.

**Review by fsuits — APPROVED** [review]
LGTM after a number of small, mostly grammar fixes.

File: docs/trestle_author.md (hunk 1 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Done

File: docs/trestle_author.md (hunk 2 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Done

File: docs/trestle_author.md (hunk 3 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Changed to YAML header

File: docs/trestle_author.md (hunk 4 above)
Comment (enikonovad) [reply docs/trestle_author.md]: done

File: docs/trestle_author.md (hunk 5 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Added missing text

File: docs/trestle_author.md (hunk 6 above)
Comment (enikonovad) [reply docs/trestle_author.md:133]: With the latest fix for duplicate headers this is not a problem anymore. If there is a duplicate, both of them will be validated

File: docs/trestle_author.md (hunk 15 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Done

File: docs/trestle_author.md (hunk 14 above)
Comment (enikonovad) [reply docs/trestle_author.md]: fixed

File: docs/trestle_author.md (hunk 13 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Yes, fixed to Validate only the headers...

File: docs/trestle_author.md (hunk 12 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Done

File: docs/trestle_author.md (hunk 11 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Done

File: docs/trestle_author.md (hunk 10 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Done

File: docs/trestle_author.md (hunk 9 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Done

File: docs/trestle_author.md (hunk 8 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Done

File: docs/trestle_author.md (hunk 7 above)
Comment (enikonovad) [reply docs/trestle_author.md]: Done
